bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/HBaseImportIO.java

```diff
@@ -0,0 +1,159 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.dataflow;
+
+import com.google.bigtable.repackaged.com.google.common.collect.ImmutableMap;
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.coders.KvCoder;
+import com.google.cloud.dataflow.sdk.io.BoundedSource;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PDone;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * {@link com.google.cloud.dataflow.sdk.Pipeline},
+ * {@link com.google.cloud.dataflow.sdk.io.Source} and
+ * {@link com.google.cloud.dataflow.sdk.transforms.PTransform}s for importing HBase Sequence Files
+ * into <a href="https://cloud.google.com/bigtable/">Google Cloud Bigtable</a>.
+ * <p>
+ * User of this class should provide dependency for
+ * {@code org.apache.hadoop.hbase.mapreduce.ResultSerialization}, which may be found in
+ * hbase-server jars.
+ *
+ * The following code snippets sets up and runs an import pipeline:
+ *
+ * <pre>
+ * {@code
+ *   HBaseImportOptions options =
+ *       PipelineOptionsFactory.fromArgs(args).withValidation().as(HBaseImportOptions.class);
+ *   Pipeline p = HBaseImportIO.createImportPipeline(options);
+ *   p
+ *       .apply("ReadSequenceFile", Read.from(HBaseImportIO.read(options)))
+ *       .apply("ConvertResultToMutations", HBaseImportIO.processRow(options))
+ *       .apply("WriteToTable", HBaseImportIO.writeToTable(options));
+ *   p.run();
+ * }
+ * </pre>
+ */
+public class HBaseImportIO {
+  private static final Logger logger = LoggerFactory.getLogger(HBaseImportIO.class);
+
+  // Needed for HBase 0.94 format. Copied from ResultSerialization.IMPORT_FORMAT_VER.
+  private static final String IMPORT_FORMAT_VER = "hbase.import.version";
+  private static final String VERSION_094_STRING = "0.94";
+
+  /**
+   * Returns a {@link Pipeline} that may write to a Cloud Bigtable.
+   */
+  public static Pipeline createImportPipeline(HBaseImportOptions options) {
+    return CloudBigtableIO.initializeForWrite(Pipeline.create(options));
```

Comment:
I don't think this is the right place for this.  If you want to make this change, it would be better to have a CloudBigtableIO.createPipeline method.